common.c

```diff
@@ -177,4 +177,65 @@ bool parseMovedReply(const char *str, NodeAddr *addr)
     /* Handle current or cluster-style -MOVED replies. */
     const char *p = strrchr(str, ' ');
     return NodeAddrParse(p + 1, strlen(p), addr);
+}
+
+/* future types should remain 4 characters or stack array in common.c/joinLinkIdleCallback() has to expand */
+char * join_type = "join";
+char * link_type = "link";
+
+/* Invoked when the connection is not connected or actively attempting
+ * a connection.
+ */
+void joinLinkIdleCallback(Connection *conn)
+{
+    char err_msg[42];
```

Comment:
I'd make it just a bit bigger to accomodate for future error message changes, etc.

Reply:
I guess its not the answer as is :)